packages/runtimes/hmr/src/loaders/hmr-runtime.js

```diff
@@ -27,8 +27,10 @@ var checkedAssets, assetsToAccept, acceptedAssets;
 // eslint-disable-next-line no-redeclare
 var parent = module.bundle.parent;
 if ((!parent || !parent.isParcelRequire) && typeof WebSocket !== 'undefined') {
-  var hostname = location.hostname;
-  var port = location.port ? ':' + location.port : '';
+  var hostname = location.origin.startsWith('http')
+    ? location.hostname
+    : 'localhost';
+  var port = location.port ? ':' + location.port : ':1234';
```

Comment:
I think we want to use the value from the CLI option here instead of hardcoding `1234`?